src/app/app.module.ts

```diff
@@ -43,11 +48,20 @@ import { HttpClientModule } from "@angular/common/http";
     ReactiveFormsModule,
     MatFormFieldModule,
     MatInputModule,
+    MatProgressSpinnerModule,
     HttpClientModule
   ],
   exports: [
     AppComponent,
   ],
+  providers: [
+      LoadingService,
+      {
+        provide: HTTP_INTERCEPTORS,
+        useClass: NetworkInterceptor,
```

Comment:
bad name for spinner interceptor

Reply:
Ok, suggest the name SpinnerInterceptor.